modules/extending_and_customizing.md

```diff
@@ -0,0 +1,155 @@
+So we may refer them as standalone blueprints or just blueprints from now on.
+
+## Basic rules for a JHipster blueprint
+
+JHipster blueprints:
+
+- are NPM packages, and are Yeoman generators.
+- follows an extension of the Yeoman rules listed at [https://yeoman.io/authoring/index.html](https://yeoman.io/authoring/index.html). Instead of being prefixed by `generator-`, it is prefixed by `generator-jhipster-`, and instead of having just the `yeoman-generator` keyword, it must have 2 keywords, `yeoman-generator` and `jhipster-blueprint`.
+
+## Usage
+
+To use a blueprint, run the below command
+
+```bash
+jhipster --blueprints <blueprint name>
+```
+
+Or use the custom provided cli:
+
+```bash
+jhipster-my-blueprint
+```
+
+### Examples
+
+JHipster has many official blueprints, some examples:
+
+- [JHipster Kotlin](https://github.com/jhipster/jhipster-kotlin) blueprint replaces most of the server side Java code with equivalent Kotlin code.
+- [JHipster.NET](https://github.com/jhipster/jhipster-dotnetcore) blueprint replaces the entire server side with .NET implementation.
+- [JHipster NodeJS](https://github.com/jhipster/generator-jhipster-nodejs) blueprint replaces the entire server side with NestJS implementation.
+- [Svelte Hipster](https://github.com/jhipster/generator-jhipster-svelte) blueprint replaces the entire client side with Svelte implementation.
+- [JHipster Native](https://github.com/jhipster/generator-jhipster-native) blueprint customizes JHipster applications with Spring Native compatibility.
+
+### Side-by-side blueprint
+
+Each generator can be a sbs blueprint. A sbs blueprint doesn't change the original generator behavior, but can customize the behavior and the result.
+Side-by-side blueprint makes easier to support multiple JHipster versions and port to a new JHipster version.
+
+To make the generator side-by-side, add this to the constructor:
+
+```js
+this.sbsBlueprint = true;
+```
+
+Example: [server generator at Native Blueprint](https://github.com/mshima/generator-jhipster-native/blob/bb9c042f6bc70a26ba8037e951c93dc1d1820983/generators/server/generator.mjs#L17).
+At this example the generator [customizes package.json](https://github.com/mshima/generator-jhipster-native/blob/bb9c042f6bc70a26ba8037e951c93dc1d1820983/generators/server/generator.mjs#L26-L35), [removes files](https://github.com/mshima/generator-jhipster-native/blob/bb9c042f6bc70a26ba8037e951c93dc1d1820983/generators/server/generator.mjs#L37-L40), [customizes pom.xml](https://github.com/mshima/generator-jhipster-native/blob/bb9c042f6bc70a26ba8037e951c93dc1d1820983/generators/server/generator.mjs#L42-L186), [customizes java files](https://github.com/mshima/generator-jhipster-native/blob/bb9c042f6bc70a26ba8037e951c93dc1d1820983/generators/server/generator.mjs#L211-L307), [customizes cypress](https://github.com/mshima/generator-jhipster-native/blob/bb9c042f6bc70a26ba8037e951c93dc1d1820983/generators/server/generator.mjs#L321-L329) and so one.
+
```

Comment:
Should these links point to `jhipster/generator-jhipster-native` instead of your fork?